core/options.c

```diff
@@ -533,20 +642,26 @@ set_dynamo_options_common(options_t *options, const char *optstr, bool for_this_
     ASSERT_OWN_OPTIONS_LOCK(options==&dynamo_options || options==&temp_options,
                             &options_lock);
     ASSERT(!OPTIONS_PROTECTED());
-    prev_pos = pos;
     while ((opt = getword(optstr, &pos, wordbuffer, sizeof(wordbuffer))) != NULL) {
         if (opt[0]=='-') {
-            /* N.B.: case 7853,7863: this expansion results in excessive stack
-             * usage under gcc.  If strcmp is intrinsic, gcc 4.1 uses 3 local
-             * slots per strcmp, and doesn't overlap them at all.  With -O2 it
-             * reduces it to 1 slot per strcmp, though it should be 0!  But
-             * gcc 3.4 only makes it intrinsic at -O1+, so -O2 fixes the
-             * problem for 4.1 but makes it appear in 3.4!  For now we
-             * explicitly use -fno-builtin-strcmp to handle all gcc versions.
-             * Long-term we may need our own strcmp inline just for this
-             * expansion that knows it needs no per-cmp slot.
-             */
-#include "optionsx.h"             // will continue if a match is found
+            value = NULL;
+            int i = 0;
+            for (i = 0; i < num_options; ++i) {
+                if (option_traits[i].type == OPTION_TYPE_bool) {
+                    set_bool_opt(opt, option_traits[i].name, &value_true,
+                                 &value_false, &value);
+                } else {
+                    set_nonbool_opt(opt, option_traits[i].name, optstr, &pos,
+                                    wordbuffer, sizeof(wordbuffer), &value);
+                }
+                if (value) {
+                    void* optptr = (void*)(options) + option_traits[i].offset;
```

Comment:
.offset is in bytes but this is adding in void* units -- gets the wrong answer, right?